src/protocol/PagingPredicateHolder.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AnchorDataListHolder} from './AnchorDataListHolder';
+import {Data} from '../serialization/Data';
+import {Predicate} from '../core/Predicate';
+import {SerializationService} from '../serialization/SerializationService';
+import {PagingPredicate} from '../serialization/DefaultPredicates';
+
+export class PagingPredicateHolder {
+    anchorDataListHolder: AnchorDataListHolder;
+    predicateData: Data;
+    comparatorData: Data;
+    pageSize: number;
+    page: number;
+    iterationTypeId: number;
+    partitionKeyData: Data;
+
+    constructor(anchorDataListHolder: AnchorDataListHolder, predicateData: Data, comparatorData: Data,
+                pageSize: number, page: number, iterationTypeId: number, partitionKeyData: Data) {
+        this.anchorDataListHolder = anchorDataListHolder;
+        this.predicateData = predicateData;
+        this.comparatorData = comparatorData;
+        this.pageSize = pageSize;
+        this.page = page;
+        this.iterationTypeId = iterationTypeId;
+        this.partitionKeyData = partitionKeyData;
+    }
+
+    static of(predicate: PagingPredicate, serializationService: SerializationService): PagingPredicateHolder {
+        // TODO PartitionPredicate is not implemented yet.
```

Comment:
What do we need to implement this feature?

Reply:
This feature is not implemented for the node.js client yet. I put that note for myself but forget to remove it.